### PR TITLE
[YUNIKORN-1948] Introduce a command to validate the content of a give…

### PR DESCRIPTION
**Pull request by yzhangal**
### What is this PR for?
When preparing the queue configuration file, manually, or use a program, a format error can happen. It can be costly to load the file to production and revert.

This Jira introduces a dedicated command to load the input queue config file (queues.yaml etc), and run the same code that production uses to validate the file, and report errors if there are any.

### What type of PR is it?
* [ ] - Bug Fix
* [X ] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-1948

### How should this be tested?
Tested locally with queues.yaml file by modifying the content to make it invalid

$ build/queueconfigchecker queues.yaml
2023-08-31T11:25:44.115-0700	ERROR	core.config	configs/config.go:181	queue configuration validation failed	{"error": "guaranteed resource of parent map[memory:3384828493824 vcore:4006191] is smaller than sum of guaranteed resources map[memory:33848285986816 vcore:4006191] of the children for queue xyz"}
github.com/apache/yunikorn-core/pkg/common/configs.ParseAndValidateConfig
	/Users/yongjunzhang/code/yunikorn/yunikorn-core/pkg/common/configs/config.go:181
github.com/apache/yunikorn-core/pkg/common/configs.LoadSchedulerConfigFromByteArray
	/Users/yongjunzhang/code/yunikorn/yunikorn-core/pkg/common/configs/config.go:154
main.main
	/Users/yongjunzhang/code/yunikorn/yunikorn-core/cmd/queueconfigchecker/queueconfigchecker.go:43
runtime.main
	/usr/local/go/src/runtime/proc.go:250

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

**Comment by yzhangal** [comment]
Hi @craigcondit Thanks for looking into. The lint issue is resolved now, but there is a codecov/project check failure, 
https://github.com/apache/yunikorn-core/pull/634/checks?check_run_id=16406753455
wonder what need to be done for this case. Thanks.

**Review by craigcondit — CHANGES_REQUESTED** [review]
In addition to the changes indicated by @pbacsko also remove the 'Visible by tests' comment  from config.go:152 on LoadSchedulerConfigFromByteArray.

File: cmd/queueconfigchecker/queueconfigchecker.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	}
+	queueFile := os.Args[1]
+	iv, err := os.ReadFile(queueFile)
+	if err != nil {
+		log.Println(err)
+		os.Exit(2)
+	}
+	_, err1 := configs.LoadSchedulerConfigFromByteArray(iv)
+	if err1 != nil {
+		log.Println(err1)
```
[the remark this answers is not in the document]
Comment (yzhangal) [reply cmd/queueconfigchecker/queueconfigchecker.go]: Many thanks @pbacsko uploaded new rev to address your comments.

**Comment by yzhangal** [comment]
> In addition to the changes indicated by @pbacsko also remove the 'Visible by tests' comment from config.go:152 on LoadSchedulerConfigFromByteArray.

Many thanks @craigcondit uploaded new rev to address your comment.

**Pull request closed without merging**